catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -155,14 +156,12 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
       SolrDocumentList docs = solrResponse.getResults();
 
       if (isFacetedQuery) {
-        List<FacetedAttributeResult> facetedAttributeResults =
-            solrResponse
-                .getFacetFields()
-                .stream()
-                .map(this::convertFacetField)
-                .collect(Collectors.toList());
-
-        responseProps.put(EXPERIMENTAL_FACET_RESULTS_KEY, (Serializable) facetedAttributeResults);
+        List<FacetField> facetFields = solrResponse.getFacetFields();
+        if (CollectionUtils.isNotEmpty(facetFields)) {
```

Comment:
If this is meant to fix the NPE, it might be better to have `getFacetFields()` return an empty list so you don't have to break the stream logic up.

Reply:
Unfortunately, `getFacetFields()` is part of Solr, so we can't change its return

Reply:
Whoops, didn't realize I had the `QueryResponse.class` file open in Intellij instead of `QueryResponse.java`